src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c

```diff
@@ -407,7 +417,7 @@ static struct drm_driver mm_drm_driver = {
 	.gem_vm_ops			= &xocl_vm_ops,
 
 	.ioctls				= xocl_ioctls,
-	.num_ioctls			= ARRAY_SIZE(xocl_ioctls),
+	.num_ioctls			= (ARRAY_SIZE(xocl_ioctls)-NUM_KERNEL_IOCTLS),
```

Comment:
What would happen if user call DRM_XOCL_EXECBUF_CB just in accident? Hope linux kernel is strong enough to handle this without crash. Have we tried it?

Reply:
The DRM subsystem in linux kernel will prevent calling of these IOCTLS. Following is a small snippet of code from drm_ioctl function (linux/v4.15/source/drivers/gpu/drm/drm_ioctl.c)
_if (is_driver_ioctl) {
		/* driver ioctl */
		if (nr - DRM_COMMAND_BASE >= dev->driver->num_ioctls)
			goto err_i1;
		ioctl = &dev->driver->ioctls[nr - DRM_COMMAND_BASE]; 
}_
These IOCTLS if called through drm_ioctl function will fail(err_i1 above). Userspace IOCTLs are invoked through drm_ioctl. These new IOCTLs can be only called using drm_ioctl_kernel, which is not exposed to userspace.  Kernel space modules need to call drm_ioctl_kernel. XSS does. Since @larry9523 also asked for this explanation, I am tagging him here.